src/main/java/org/skriptlang/skriptworldguard/elements/expressions/ExprRegionFlag.java

```diff
@@ -0,0 +1,131 @@
+package org.skriptlang.skriptworldguard.elements.expressions;
+
+
+
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+import com.sk89q.worldguard.WorldGuard;
+import com.sk89q.worldguard.protection.flags.*;
+
+import com.sk89q.worldguard.protection.regions.ProtectedRegion;
+
+
+import org.bukkit.event.Event;
+import org.skriptlang.skriptworldguard.SkriptWorldGuard;
+import org.skriptlang.skriptworldguard.worldguard.WorldGuardRegion;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ExprRegionFlag extends PropertyExpression<WorldGuardRegion, String> {

```

Comment:
I will prepare the class to work with a ClassInfo for the Flag type then. If I have time I will create the Flag type aswell

Reply:
Would setting the return type to a WorldGuardFlag classinfo mean that it will also only accept WorldGuardFlag as changers?

If so, wouldn't this make it "harder" to manage flags?

set flag "greeting" of region %worldguardregion% to %worldguardflag% looks repetitive.

Also, while creating the classinfo, what's the difference between using the normal Flag.class from WorldGuard and creating a WorldGuardClass.class with getters to access its Flag class inside?

I saw that this 2nd way is used for the WorldGuardRegion class for example. Why this double step?

Reply:
No, region flags accept various types of data (boolean/string/numbers/etc.) so you may need to either accept Object.class or accept all types mentioned [here](https://worldguard.enginehub.org/en/latest/regions/flags/)

You will then have to do the proper checking and casting for `change` method

As for creating a wrapper class for Flag, if that helps a lot I think it's fine, but if it's not really needed then we better stick with WG Flag class

Reply:
Oh I think my comment here never posted...

the WorldGuardRegion wrapper exists because ProtectionRegion seemingly does not provide any details as to the world it is located in

Reply:
Gotcha, tysm guys!

Reply:
I am a little unsure about a flag classinfo, mainly for the fact that it may be rather large + I'm not sure how it might handle WorldGuard addons that add new flags